HerPublicWebsite.BusinessLogic/ExternalServices/OsPlaces/OsPlacesApi.cs

```diff
@@ -1,33 +1,52 @@
+using HerPublicWebsite.BusinessLogic.Extensions;
+using HerPublicWebsite.BusinessLogic.ExternalServices.Common;
 using HerPublicWebsite.BusinessLogic.Models;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 
 namespace HerPublicWebsite.BusinessLogic.ExternalServices.OsPlaces;
 
 public class OsPlacesApi : IOsPlacesApi
 {
-    public async Task<List<Address>> GetAddressesAsync(string postcode, string buildingNumber)
+    private readonly OsPlacesConfiguration config;
+    private readonly ILogger<OsPlacesApi> logger;
+
+    public OsPlacesApi(IOptions<OsPlacesConfiguration> options, ILogger<OsPlacesApi> logger)
+    {
+        this.config = options.Value;
+        this.logger = logger;
+    }
+    
+    public async Task<List<Address>> GetAddressesAsync(string postcode, string buildingNameOrNumber)
     {
-        //TODO BEISHER-248: Actually implement API here
-        return new List<Address> {
-            new () {
-                DisplayAddress = "82 My House, My Area, A Special Location, LB6 420",
-                AddressLine1 = "82 My House",
-                AddressLine2 = "My Area",
-                Postcode = "LB6 420",
-                County = "A County",
-                Town = "Camdoon",
-                Uprn = "023232323",
-                LocalCustodianCode = "2323",
-            },
-            new ()
+        if (!postcode.IsValidUkPostcodeFormat())
+        {
+            return new List<Address>();
+        }
+        
+        var parameters = new RequestParameters
+        {
+            BaseAddress = config.BaseUrl,
+            Path = $"/search/places/v1/postcode?postcode={postcode}",
```

Comment:
Yes, that's what I was hoping. Requesting LPI as well adds extra complexities (e.g. one LPI search returned a result that was a street rather than a property). Maybe we should raise a ticket to look at it if we have time.